### PR TITLE
Port 10714 bug none type object has no attribute get error during resync in new relic integration

### PR DESCRIPTION
**Pull request by shariff-6**
# Description

# Description

### What
This PR fixes a bug where a `NoneType` object error (`'NoneType' object has no attribute 'get'`)  is thrown 

### Why
The issue occurs due to missing checks for `None` responses in `service_levels.py`, causing the resync to fail when attempting to access attributes on a `NoneType` object.

### How
- Added  checks to ensure the response is not `None` before accessing attributes.
- Updated logging to capture and warn about empty or invalid responses.
- Refactored related methods to handle edge cases gracefully.
- Updated unit tests to include scenarios with `None` or empty responses.

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/newrelic/newrelic_integration/core/service_levels.py [hunk 1]
```diff
@@ -22,6 +23,10 @@ def __init__(self, http_client: httpx.AsyncClient):
     async def get_service_level_indicator_value(
         self, http_client: httpx.AsyncClient, nrql: str
     ) -> dict[Any, Any]:
+        if not nrql:
+            logger.debug("Empty NRQL provided")
```
Comment (PeyGis) [inline integrations/newrelic/newrelic_integration/core/service_levels.py:27]:
```suggestion
            logger.debug("Empty NRQL provided, skipping graph api request...")
```

File: integrations/newrelic/newrelic_integration/core/service_levels.py [hunk 2]
```diff
@@ -30,28 +35,32 @@ async def get_service_level_indicator_value(
         response = await send_graph_api_request(
             http_client, query, request_type="get_service_level_indicator_value"
         )
+
+        if not response:
+            logger.warning("Empty response from API")
```
Comment (PeyGis) [inline integrations/newrelic/newrelic_integration/core/service_levels.py:40]:
```suggestion
            logger.warning("Empty response from API, skipping...")
```

File: integrations/newrelic/newrelic_integration/core/service_levels.py [hunk 3]
```diff
@@ -30,28 +35,32 @@ async def get_service_level_indicator_value(
...
-        nrql = (
-            service_level.get("serviceLevel", {})
-            .get("indicators", [])[0]
-            .get("resultQueries", {})
-            .get("indicator", {})
-            .get("nrql")
-        )
+        indicators = service_level.get("serviceLevel", {}).get("indicators", [])
+        if not indicators:
+            logger.warning("No indicators found in service level")
```
Comment (PeyGis) [inline integrations/newrelic/newrelic_integration/core/service_levels.py:58]:
```suggestion
            logger.warning("No indicators found in service level, returning the raw service level")
```

**Review by PeyGis — APPROVED** [review]
lgtm

File: integrations/newrelic/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
+## 0.2.00 (2024-11-28)
```
Comment (Tankilevitch) [inline integrations/newrelic/CHANGELOG.md:10]: 0.1.100

**Review by Tankilevitch — APPROVED** [review]
